### ISSUE TITLE
How about including [fish](https://fishshell.com)? I started a [project](https://github.com/etcusrvar/fish-direnv-completions) since I haven't found another like it. direnv already supports a hook for fish.

### ISSUE THREAD

**Issue by zimbatm** [direnv/direnv#753]
How about including [fish](https://fishshell.com)? I started a [project](https://github.com/etcusrvar/fish-direnv-completions) since I haven't found another like it. direnv already supports a hook for fish.

_Originally posted by @etcusrvar in https://github.com/direnv/direnv/issues/443#issuecomment-767999744_

**Comment by zimbatm** [comment]
Sounds good! I would be happy to accept a PR for fish completion.

**Comment by etcusrvar** [comment]
Thanks @zimbatm. Let me work on making the completions I started a bit more robust, then we can work on merging into direnv.

**Comment by mrmeszaros** [comment]
@etcusrvar I would be glad to help out if I can!
I see that the linked  [project etcusrvar/fish-direnv-completions](https://github.com/etcusrvar/fish-direnv-completions) is superseded by your [fish-completions](https://github.com/etcusrvar/fish-completions).

Also, I noticed that the wording of the help for commands is a bit messy - like `allow`, `deny`, `hook` and `help`.
I suggest to use a unified style - like the one git uses.
For example some rework could be:
```
allow [PATH]
    Grant authorization to load the given .envrc
deny [PATH]
    Revoke authorization to load the given .envrc
hook [SHELL]
    Setup the shell hook
help [--show-private]
    Show this help
```

**Comment by etcusrvar** [comment]
@mrmeszaros yeah I've moved various one-off projects around quite a bit. I've gotten into other tasks recently and haven't looked at fish completions in a while. Feel free to take what I started and improve it and turn it into a PR if you have the bandwidth.
